Remove obsolete resources caused by merging controllers

aliok opened the pull request:
<!--
Request Prow to automatically lint any go code in this PR:

/lint
-->

# Issue to be fixed

Fixes #90

## Proposed Changes

* Remove old resources that are obsolote after https://github.com/knative/eventing/pull/2448

## Release Note

<!-- Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". -->

```release-note
NONE
```

More info on how to find what went obsolete is available at https://github.com/knative/eventing-operator/issues/90

How to verify:
- Install the 0.12.0 operator, create a CR
- replace that operator with this PR
- the obsolote resources should be deleted, new resources should be created

Notes:
- Replaced the `0.12.0.yaml` in `kodata` with the nightly release file (`v20200206-05bba2be.yaml`)
- That nightly release file is fetched like this: `gsutil cp gs://knative-nightly/eventing/latest/eventing.yaml ./`

aliok commented:
cc @houshengbo

aliok commented:
Removed the WIP status to see what tests do, but can somebody please add a `/hold`?
cc @houshengbo @k4leung4

houshengbo commented:
/ok-to-test

houshengbo commented on pkg/reconciler/knativeeventing/knativeeventing.go:
```diff
@@ -207,3 +209,51 @@ func (r *Reconciler) updateStatus(desired *eventingv1alpha1.KnativeEventing) (*e
+
+	resource.SetNamespace("knative-eventing")
+
+	// 0.13
```
I suggest you change the comment to `remove old resources at 0.12`, because 0.13 can make us confused that these resources are in 0.13

aliok replied:
Done, thx for the suggestion

houshengbo commented:
@aliok when we update the manifest, we need to update the version as well to match in version.go

aliok commented:
/test pull-knative-eventing-operator-integration-tests-on-latest-eventing

aliok commented:
@houshengbo I addressed your comments. Thanks for the feedback.

As I said in https://github.com/knative/eventing-operator/pull/93, let's merge this PR first and then that one. See that PR for my reasons.
